src/components/v2/Modal/ModalStyles.ts

```diff
@@ -0,0 +1,54 @@
+import { useTheme } from '@mui/material';
+
+const iconCloseSize = '35px';
+
+export const useModalStyles = ({ hasTitleComponent }: { hasTitleComponent: boolean }) => {
+  const theme = useTheme();
+  return {
+    box: {
+      position: 'absolute' as const,
+      top: '50%',
+      left: '50%',
+      transform: 'translate(-50%, -50%)',
+      width: 400,
+      bgColor: 'background.paper',
+      border: '2px solid #000',
+      boxShadow: 24,
+      padding: 0,
+      borderRadius: '24px',
+      paddingTop: '24px',
+    },
+    titleWrapper: {
+      position: 'relative',
+      paddingLeft: '24px',
+      paddingRight: '24px',
+      paddingBottom: hasTitleComponent ? '24px' : 0,
+      borderBottom: hasTitleComponent ? `1px solid ${theme.palette.secondary.dark}` : 0,
+      marginBottom: hasTitleComponent ? '24px' : 0,
+      paddingTop: 0,
+    },
+    titleComponent: {
+      alignSelf: 'center',
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'center',
+      minHeight: iconCloseSize,
+      paddingLeft: iconCloseSize,
+      paddingRight: iconCloseSize,
+    },
+    closeIcon: {
+      right: '24px',
+      top: 0,
+      position: 'absolute',
+      height: iconCloseSize,
+      width: iconCloseSize,
+      minWidth: iconCloseSize,
+      marginLeft: 'auto',
+      padding: 0,
+    },
+    contentWrapper: {
+      padding: '24px',
```

Comment:
I think we should have an optional prop like `noHorizontalPadding` that can be passed to the component to remove the left and right paddings, otherwise we'll have issues building this modal:
<img width="525" alt="Screenshot 2022-03-25 at 10 33 01" src="https://user-images.githubusercontent.com/44675210/160104509-c5318e3b-d931-418c-b7dc-aa9ea8e5705a.png">